staff/ismael-haytam/pre-course/tema-1/Methods.js

```diff
@@ -0,0 +1,335 @@
+
+/*
+    -   Methods ---------------------------------------------------------
+*/
+
+
+// a) Puedes contar cuantas letras tiene tu nombre?
+
+let countString = name => console.log(name.split(' ').join('').length);
+
+countString('Ismael Haytam');
+
+
+
+
+// b) Añade tu apellido e indica en que posición del string empieza.
+
+let surnamePosition = fullName => fullName.split(' ').forEach( (s, i) => (s === 'Tanane') ? console.log(i + 1) : '');
+
+surnamePosition('Ismael Haytam Tanane');
+
+
+
+
+
+// c & d1 ) Ahora, con tu apellido y nombre en la misma variable, muestra solo el nombre:
+
+function printName(fullName) {
+    let arrayNames = fullName.split(' ');
+    let name = arrayNames[0] + ' ' + arrayNames[1];
+    return console.log(name);
+}
+
+printName('Ismael Haytam');
+
+
+
+
+
+// d) Ahora, solo tu apellido.
+
+let printSurname = fullName => console.log(fullName.split(' ')[2]);
+
+printSurname('Ismael Haytam Tanane');
+
+
+
+
+
+// e) Ahora, reemplaza tu nombre por "Mr/Ms" y vuelve a mostrar la variable con los cambios.
+
+let salute = fullName => console.log(`Hello, ${fullName.replace('Ismael Haytam', 'Mr.')}`);
+
+salute('Ismael Haytam Tanane');
+
+
+
+
+
+// f) Selecciona tu apellido y transfórmalo a MAYÚSCULAS.
+
+let surnameToUppercase = (fullName) => console.log(fullName.split(' ')[2].toUpperCase());
+
+surnameToUppercase('Ismael Haytam Tanane');
+
+
+
+
+
+// g) Ahora declara una variable nueva e igualala a la anterior variable sumándole, además, un mensaje
+
+function awesome(fullName) {
+    let arrayames = fullName.split(' ');
+    arrayames[2] = 'is awesome';
+    return console.log(arrayames.join(' '));
+}
+
+awesome('Ismael Haytam Tanane');
+
+
+
+
+
+// h) Ahora, puedes seleccionar la inicial de tu nombre y apellido y mostrarlas por pantalla?
+
+function initialOfTheFullName (fullName) {
+    let initials = [];
+    fullName.split(' ').forEach( s => initials.push(s.substr(0, 1).toUpperCase()));
+    return console.log(initials.join('.'));
+}
+
+initialOfTheFullName('Ismael Haytam Tanane');
+
+
+
+
+
+/*
+    -   Arrays ---------------------------------------------------------
+*/
+
+// a) Declara tu nombre completo en una array y muéstralo por pantalla separando cada letra por "/"
+
+let separateLettersBySlash = fullName  => console.log(fullName.split(' ').join('').toUpperCase().split('').join('/'));
+
+separateLettersBySlash('Ismael Haytam Tanane');
+
+
+
+
+
+// b) Ahora solo selecciona tu apellido y muestra cada letra separada por "|"
+
+let separateSurnameBySlash = fullName =>  console.log(fullName.split(' ')[2].toUpperCase().split('').join('/'));
+
+separateSurnameBySlash('Ismael Haytam Tanane');
+
+
+
+
+
+// c) Ahora muestra cada letra de tu nombre con su posición
+
+function positionOfEachLetterOfTheName (fullName) {
+    let result = [];
+    fullName = fullName.replace('Tanane', '');
```

Comment:
do not re-assign values to argument references (fullName), create new local references instead